### PR TITLE
Added requestedTenant access info in User model

### PR DESCRIPTION
**Pull request by vamsi-amazon**
Signed-off-by: vamsi-amazon <reddyvam@amazon.com>

### Description
Added requestedTenancy access info in User Model.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by getsaurabh02 — COMMENTED** [review]
Which release are planning to release this? Also do we have any backport plans considering this could be a breaking change for the consumer plugins.

**Comment by vamsi-amazon** [comment]
> Which release are planning to release this? Also do we have any backport plans considering this could be a breaking change for the consumer plugins.

I want to backport to all the versions till 1.3. 

I am not sure of all the consumers of the plugin. If the consumers are creating User then this would cause issues if they don't upgrade because of the changes in constructor. 

Should I keep the old constructor as well? I guess this is the safest way so that whoever is using old constructor would not cause any failures.

**Comment by dblock** [comment]
> Should I keep the old constructor as well? I guess this is the safest way so that whoever is using old constructor would not cause any failures.

Yes, that's probably the right thing to do.

File: src/main/java/org/opensearch/commons/authuser/User.java [hunk 1]
```diff
@@ -62,20 +66,38 @@ public User(final String name, final List<String> backendRoles, List<String> rol
...
         this.customAttNames = customAttNames;
         this.requestedTenant = null;
+        this.requestedTenantAccess = null;
+    }
+
+    public User(
+        final String name,
+        final List<String> backendRoles,
+        List<String> roles,
+        List<String> customAttNames,
```
Comment (getsaurabh02) [inline src/main/java/org/opensearch/commons/authuser/User.java:76]: nit: final

**Review by getsaurabh02 — CHANGES_REQUESTED** [review]

**Review by getsaurabh02 — APPROVED** [review]

File: src/main/java/org/opensearch/commons/authuser/User.java [hunk 2]
```diff
@@ -40,20 +40,24 @@ final public class User implements Writeable, ToXContent {
...
+    public static final String REQUESTED_TENANT_ACCESS = "user_requested_tenant_access";
 
     private final String name;
     private final List<String> backendRoles;
     private final List<String> roles;
     private final List<String> customAttNames;
     @Nullable
     private final String requestedTenant;
+    @Nullable
+    private final String requestedTenantAccess;
```
Comment (peternied) [inline src/main/java/org/opensearch/commons/authuser/User.java:52]: Instead of treating this as a string can we make it an enum?

**Pull request closed without merging**